### ISSUE TITLE
2019-04-RC1 Norne Fails Due to Info Messages Limit

### ISSUE THREAD

**Issue by OPMUSER** [OPM/opm-simulators#1794]
Norne model fails because the INFO message limit exceeds the default value.
```
Error summary:
Warnings          281
**Info              10118**
Errors            0
Bugs              0
Debug             0
Problems          2
```
This is because OPM Flow is still writing out the well control change during a TSTEP iteration, 
```
Time step 0, stepsize 16 days.
    Switching control mode for well B-2H from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well B-2H from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well B-2H from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well B-2H from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well B-2H from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well B-2H from RESERVOIR_RATE to SURFACE_RATE
Time step summary: well its =  4, newton its =  4, linearizations =  5 ( 0.634 sec), linear its =  76 ( 1.285 sec)

```
as oppose to writing the message only if it changes from the previous TSTEP. See  #1703

**Comment by GitPaean** [comment]
```
    Switching control mode for well B-2H from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well B-2H from RESERVOIR_RATE to SURFACE_RATE
``` 

It is something I never understood. At least it needs to have some switching from SURFACE_RATE to RESERVOIR_RATE .  There is something pretty fishy there. 

> as oppose to writing the message only if it changes from the previous TSTEP

Yes. It remains to be done.

**Comment by OPMUSER** [comment]
Still the same problem with RC2:

```
   Report step 246/247 at day 3298/3312, date = 17-Nov-2006
   Time step 0, stepsize 14 days.
    Switching control mode for well B-2H from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well D-2H from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well B-3H from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well B-4DH from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well E-3CH from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well E-2AH from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well B-1BH from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well B-2H from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well D-2H from RESERVOIR_RATE to SURFACE_RATE
    Switching control mode for well B-3H from RESERVOIR_RATE to SURFACE_RATE
...............................
```
Is this because the the WCONHIST keyword is set to RESV? So why is OPM Flow switching to SURFACE_RATE. If OPM Flow is converting from RESV to surface volumes then we don't need this message.

**Comment by atgeirr** [comment]
I have identified the cause of this issue. I am not sure we can have a fix in for the release.

**Comment by OPMUSER** [comment]
@atgeirr Well that is good news and even better news if we can get the fix into the release. I say this because if users have this problem on their own models then it looks like a regression from the users point of view, so it would be good to have this fixed. 

I know, easy for me to say.

**Comment by atgeirr** [comment]
Just a note: even if the info messages are suppressed due to going over the limit, the simulator does not crash, and continues to produce output, just not to the terminal (so it might look like it is hanging).

I am testing the fix now.

**Comment by atgeirr** [comment]
Fix is in #1809. I am not sure it will fix everything though, as I do not think I ever got your sky-high info message counts, on a relatively recent Norne run (early April) I see a report like this:
```
Error summary:
Warnings          281
Info              1658
Errors            0
Bugs              0
Debug             4121
Problems          4
```

whereas an older run from before the logging change that introduced the bug (December) gives:

```
Error summary:
Warnings          281
Info              1099
Errors            0
Bugs              0
Debug             3702
Problems          2
```

In any case, nowhere near 10k.

**Comment by OPMUSER** [comment]
I just ran the Norne model with the default parameter set and RC2 and got this:

```
Error summary:
Warnings          281
Info              10118
Errors            0
Bugs              0
Debug             0
Problems          2
```
So may be the difference is with the parameter set, but I'll wait until RC3 ;-) is out before commenting again.

**Comment by atgeirr** [comment]
It turns out that the current master as well as the release candidate is broken with the Norne case, and the 10k info messages is just a symptom of that! We have found the likely culprit, and are fixing ASAP.

**Comment by atgeirr** [comment]
Fix in opm-common#746 for the main problem (incorrect switching to ORAT control in history mode), #1809 is also good for eliminating incorrectly repeated log messages.

Thanks for not giving up on this, we only today realized what was the root cause.

**Comment by OPMUSER** [comment]
Great, glad we got to the bottom of this and got the patch in the release branch.
